### DISCUSSION TITLE
INFERENCE WITH EXISTING MODELS - GOOGLE colab - issues

### DISCUSSION THREAD

**Discussion by mary1204** [open-mmlab/mmpose#2801] [Q&A]
Hello!

I am trying to install mmpose to further be able to detect the 3d pose estimation of the objects of my own videos.
I am working on colab. I started by installing the mmpose based on the steps in the section: installation. After that, I went to the section: INFERENCE WITH EXISTING MODELS, and when trying to do the first code of that section, I am facing some errors.

The code I am using is: 

from mmpose.apis import MMPoseInferencer

img_path = '/content/mmpose/tests/data/coco/000000000785.jpg'   # replace this with your own image path

# instantiate the inferencer using the model alias
inferencer = MMPoseInferencer('human')

# The MMPoseInferencer API employs a lazy inference approach,
# creating a prediction generator when given input
result_generator = inferencer(img_path, show=True)
result = next(result_generator)

And the error I am facing is this one:

"11/09 16:31:09 - mmengine - WARNING - The current default scope "mmdet" is not "mmpose", `init_default_scope` will force set the currentdefault scope to "mmpose".
Loads checkpoint by http backend from path: https://download.openmmlab.com/mmpose/v1/projects/rtmposev1/rtmpose-m_simcc-body7_pt-body7_420e-256x192-e48f03d0_20230504.pth
11/09 16:31:10 - mmengine - WARNING - The current default scope "mmpose" is not "mmdet", `init_default_scope` will force set the currentdefault scope to "mmdet".
Loads checkpoint by http backend from path: https://download.openmmlab.com/mmdetection/v3.0/rtmdet/rtmdet_m_8xb32-300e_coco/rtmdet_m_8xb32-300e_coco_20220719_112220-229f527c.pth
Inference ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   
The model and loaded state dict do not match exactly

unexpected key in source state_dict: data_preprocessor.mean, data_preprocessor.std

/usr/local/lib/python3.10/dist-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming 
release, it will be required to pass the indexing argument. (Triggered internally at 
../aten/src/ATen/native/TensorShape.cpp:3526.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]"

And then the session crashed.

Can someone help me with this problem? Thank you a lot!

**Comment by Ben-Louis** [comment]
Setting `show=True` only works in environments with GUI. Consider setting `vis_out_dir` and visualize output image manually with opencv. Refer to https://mmpose.readthedocs.io/en/latest/user_guides/inference.html#arguments-of-inferencer for info on inferencer arguments.
